spec/ndx-ophys-devices.extensions.yaml

```diff
@@ -1,8 +1,216 @@
 groups:
-- neurodata_type_def: TetrodeSeries
-  neurodata_type_inc: ElectricalSeries
-  doc: An extension of ElectricalSeries to include the tetrode ID for each time series.
+- neurodata_type_def: Indicator
+  neurodata_type_inc: Device
+  doc: Extends Device to hold metadata on the Indicator.
   attributes:
-  - name: trode_id
-    dtype: int32
-    doc: The tetrode ID.
+  - name: label
+    dtype: text
+    doc: Indicator standard notation.
+  - name: injection_location
+    dtype: text
+    doc: Injection brain region name.
+    required: false
+  - name: injection_coordinates_in_mm
+    dtype: float
+    shape:
+    - 3
+    doc: Indicator injection location in stereotactic coordinates (AP, ML, DV) mm
+      relative to Bregma.
+    required: false
+- neurodata_type_def: OpticalFiber
+  neurodata_type_inc: Device
+  doc: Extends Device to hold metadata on the Optical Fiber.
+  attributes:
+  - name: model
+    dtype: text
+    doc: Model of optical fiber.
+    required: false
+  - name: numerical_aperture
+    dtype: float
+    doc: Numerical aperture, e.g., 0.39 NA.
+    required: false
+  - name: core_diameter_in_um
+    dtype: float
+    doc: Core diameter in micrometers.
+    required: false
+- neurodata_type_def: ExcitationSource
+  neurodata_type_inc: Device
+  doc: Extends Device to hold metadata on the Excitation Source.
+  attributes:
+  - name: model
+    dtype: text
+    doc: Model of excitation source device.
+    required: false
+  - name: illumination_type
+    dtype: text
+    doc: 'Type of illumination. Suggested values: LED, LAser or more specifically
+      Gas, Metal vapor, Solid state, Dye, Semiconductor, Free electron, etc.'
+  - name: excitation_wavelength_in_nm
+    dtype: float
+    doc: Excitation wavelength of the stimulation light (nanometers).
+  - name: peak_power_in_W
+    dtype: float
+    doc: Incident power of stimulation device (in Watts).
+    required: false
+  - name: peak_pulse_energy_in_J
+    dtype: float
+    doc: If device is pulsed light source, pulse energy (in Joules).
+    required: false
+  - name: intensity_in_W_per_m2
+    dtype: float
+    doc: Intensity of the excitation in W/m^2, if known.
+    required: false
+  - name: pulse_rate_in_Hz
+    dtype: float
+    doc: If device is pulsed light source, pulse rate (in Hz) used for stimulation.
+    required: false
+  - name: exposure_time_in_s
+    dtype: float
+    doc: Exposure time of the sample (in sec).
+    required: false
+- neurodata_type_def: Photodetector
+  neurodata_type_inc: Device
+  doc: Extends Device to hold metadata on the Photodetector.
+  attributes:
+  - name: model
+    dtype: text
+    doc: Model of photodetector device.
+    required: false
+  - name: detector_type
+    dtype: text
+    doc: 'Technology used to detect the light. Suggested values: CCD, Intensified
+      CCD, Analog video, PMT, Photodiode, CMOS, EBCCD, FTIR.'
```

Comment:
Analog video as a photodetector? 🤔 What is that like?

Reply:
Actually, I am not so sure; I included it following the same examples reported for the detector type in the [openmicroscopy standard](. https://brainportal.s3.amazonaws.com/3D-MMS/Instrument_March_22_2022.pdf). I will look into it.

Reply:
lol OK; seems possible I suppose, just never seen or heard anyone do it before